### PR TITLE
add Wasm extension feature stage.

### PR DESCRIPTION
**Pull request by bianpengyuan**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by duderino — APPROVED** [review]
Thanks for doing this Peter.

**Review by douglas-reid — APPROVED** [review]

**Review by duderino — CHANGES_REQUESTED** [review]
Actually, @bianpengyuan I think this should be experimental because it still relies on EnvoyFilter and that's not really an API.

@mandarjog @douglas-reid @PiotrSikora was something else agree on?

**Comment by bianpengyuan** [comment]
@duderino Yeah make sense, we should block Wasm extension distribution alpha on API. I have updated this PR to mark this feature as experimental.  I thought alpha is the `initial` stage of any feature, but actually alpha also needs promotion. but I keep overall Wasm feature status as `alpha`, which is what we claimed at Istio 1.5.

File: content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ owner: istio/wg-policies-and-telemetry-maintainers
 test: no
 ---
 
+{{< boilerplate experimental-feature-warning >}}
```
Comment (ericvn) [inline content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md:13]: Normally [Experimental] is added to the end of the title as well.

File: content/en/about/feature-stages/index.md [hunk 2]
```diff
@@ -59,6 +59,12 @@ Below is our list of existing features and their current phases. This informatio
 | [Trace Sampling](/docs/tasks/observability/distributed-tracing/configurability/#trace-sampling) | Beta
 | [Request Classification](/docs/tasks/observability/metrics/classify-metrics/) | Beta
 
+### Extensibility
+
+| Feature           | Phase
+|-------------------|-------------------
+| [WebAssembly Extension](/docs/ops/configuration/extensibility/wasm-module-distribution) | Alpha
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:66]: I does seem strange to have an Alpha feature have a link that leads to an Experimental page.

File: content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md (hunk 1 above)
Comment (bianpengyuan) [reply content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md:13]: Done.

File: content/en/about/feature-stages/index.md (hunk 2 above)
Comment (bianpengyuan) [reply content/en/about/feature-stages/index.md]: I removed the doc link. We don't really have any other doc about Wasm except this linked one, or a blog post created back in 1.5 where we claimed alpha support for Wasm. Looks like there are other features in this page which does not have link attached, so I simply remove the link.

**Review by duderino — APPROVED** [review]

**Comment by duderino** [comment]
TOC previously approved Wasm at Alpha, so I think that's valid.   Introducing Wasm remote loading as a separate feature at experimental also lgtm.  Splitting the pre-existing support/maturity from the new feature nicely addresses the issue.

**Review by ericvn — APPROVED** [review]

**Pull request merged**